### PR TITLE
[MSHARED-893] deprecate more file methods we don't need in Java 7+

### PR DESCRIPTION
**Pull request by elharo**
@michael-o

### CODE REVIEW

File: src/main/java/org/apache/maven/shared/utils/io/FileUtils.java [hunk 1]
```diff
@@ -573,10 +575,10 @@ private static boolean isValidFile( @Nonnull String file, @Nonnull String... ext
...
     /**
-     * Simple way to make a directory
+     * Simple way to make a directory.
      *
      * @param dir the directory to create
-     * @throws IllegalArgumentException if the dir contains illegal Windows characters under Windows OS.
+     * @throws IllegalArgumentException if the dir contains illegal Windows characters under Windows OS
      * @see #INVALID_CHARACTERS_FOR_WINDOWS_FILE_NAME
      */
     public static void mkdir( @Nonnull String dir )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:584]: Why didn't you deprecate this one?

File: src/main/java/org/apache/maven/shared/utils/io/FileUtils.java [hunk 2]
```diff
@@ -784,7 +792,9 @@ private static void copyFileToDirectoryIfModified( @Nonnull final File source,
      * @throws IOException  if <code>source</code> does not exist, <code>destination</code> cannot be
      *                      written to, or an IO error occurs during copying
      * @throws java.io.FileNotFoundException if <code>destination</code> is a directory
+     * @deprecated use {@code org.apache.commons.io.FileUtils.copyFile()}
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:795]: What about `Files#copy(Path, Path, Option...)`.

File: src/main/java/org/apache/maven/shared/utils/io/FileUtils.java [hunk 3]
```diff
@@ -913,7 +923,9 @@ public static void copyURLToFile( @Nonnull final URL source, @Nonnull final File
      *                     <li><code>destination</code> cannot be written to</li>
      *                     <li>an IO error occurs during copying</li>
      *                     </ul>
+     * @deprecated use {@code org.apache.commons.io.FileUtils.copyInputStreamToFile()}
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:926]: Available in `Files` too.

File: src/main/java/org/apache/maven/shared/utils/io/FileUtils.java (hunk 1 above)
Comment (elharo) [reply src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:584]: This one has some weird logic for handling file names on Windows I'm not sure is available elsewhere.

File: src/main/java/org/apache/maven/shared/utils/io/FileUtils.java (hunk 2 above)
Comment (elharo) [reply src/main/java/org/apache/maven/shared/utils/io/FileUtils.java]: good idea, done

File: src/main/java/org/apache/maven/shared/utils/io/FileUtils.java (hunk 3 above)
Comment (elharo) [reply src/main/java/org/apache/maven/shared/utils/io/FileUtils.java]: done

File: src/main/java/org/apache/maven/shared/utils/io/FileUtils.java (hunk 1 above)
Comment (michael-o) [reply src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:584]: Won't Files#createDirectory()` properly fail in such a case? If so, I would rather rely on this one.

Comment (elharo) [reply src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:586]: OK

**Comment by jira-importer** [comment]
Resolve #288

**Pull request merged**